crates/matrix-sdk-common/src/linked_chunk/lazy_loader.rs

```diff
@@ -196,6 +196,59 @@ where
     Ok(())
 }
 
+/// A pretty inefficient, test-only, function to rebuild a full `LinkedChunk`.
+#[doc(hidden)]
+pub fn from_all_chunks<const CAP: usize, Item, Gap>(
+    mut chunks: Vec<RawChunk<Item, Gap>>,
+) -> Result<Option<LinkedChunk<CAP, Item, Gap>>, LazyLoaderError>
+where
+    Item: Clone,
+    Gap: Clone,
+{
+    if chunks.is_empty() {
+        return Ok(None);
+    }
+
+    // Sort by `next` so that the search for the next chunk is faster (it should
+    // come first). The chunk with the biggest next chunk identifier comes first.
+    // Chunk with no next chunk comes last.
+    chunks.sort_by(|a, b| b.next.cmp(&a.next));
+
+    let last_chunk = chunks.pop().unwrap();
```

Comment:
I know that this is meant to be a test function, but it isn't marked as `cfg(test)` and those unneeded unwraps bother me.

Why not replace the `is_empty()` check with a:

```rust
let Some(last_chunk) = vec.last() else {
     return Ok(None)
};
```

Reply:
It's not behind a `cfg(test)` because it's used by other crates :-/.

I can do a `pop() else` yup.

Reply:
Actually the `pop` cannot fail because `chunks` is guaranteed to not be empty. I am adding a `// SAFETY` comment.